Index: src/libLastFmTools/containerutils.cpp
===================================================================
--- src/libLastFmTools/containerutils.cpp	(revision 34972)
+++ src/libLastFmTools/containerutils.cpp	(working copy)
@@ -67,7 +67,7 @@
 
 QString dataPath( QString file )
 {
-    return QApplication::applicationDirPath() + "/data/" + file;
+    return "/usr/share/lastfm/" + file;
 }
 
 
@@ -119,7 +119,7 @@
         // Hack to get it working with VS2005
         dirPath = qApp->applicationDirPath();
     #else
-        dirPath = qApp->applicationDirPath() + "/services";
+        dirPath = "/usr/lib/lastfm/";
     #endif
 
     #ifndef QT_NO_DEBUG

